File: packages/contracts/v1/project.ts
import type { UUID, ISO8601 } from "./common.js";

export interface Project {
  schema_version: "1.0";
  id: UUID;
  name: string;
  owner_id: UUID;
  status: "draft" | "processing" | "ready" | "error";
  floorplan_asset_id: UUID | null;
  inspiration_board_id: UUID | null;
  canonical_model_id: UUID | null;
  metadata: Record<string, unknown>;
  created_at: ISO8601;
  updated_at: ISO8601;
}

File: packages/contracts/python/noa_contracts/project.py
from __future__ import annotations

from typing import Literal
from pydantic import BaseModel


class Project(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    name: str
    owner_id: str
    status: Literal["draft", "processing", "ready", "error"]
    floorplan_asset_id: str | None
    inspiration_board_id: str | None
    canonical_model_id: str | None
    metadata: dict[str, object] = {}
    created_at: str
    updated_at: str
